DogTales/dog/player/player_controller.cpp

```diff
@@ -0,0 +1,45 @@
+#include <dog/player/player_controller.hpp>
+#include <algorithm>
+#include <cassert>
+
+namespace dog {
+
+PlayerController::PlayerController(bave::NotNull<bave::App const*> app) : m_app(app) {}
+
+void PlayerController::bind_throttle(std::string_view const id, Throttle const throttle) {
+	if (throttle.hi == bave::Key::eUnknown) { return; }
+	auto it = m_mappings.find(id);
+	if (it == m_mappings.end()) {
+		auto [i, _] = m_mappings.insert_or_assign(std::string{id}, std::vector<Mapping>{});
+		it = i;
+	}
+	assert(it != m_mappings.end());
+	it->second.push_back(throttle);
+}
+
+void PlayerController::bind_key(std::string_view id, bave::Key key) {
```

Comment:
so, `Throttle` is specifically for keys that control a range? if we were making a game with no movement and only jumping, we could theoretically bypass the throttle binding altogether?

Reply:
Yeah, couldn't think of a better name. 😅
And yup, if a game only needs single binary input states (0 or 1), there's no real need to associate a pair of keys to a binding.